### PR TITLE
Add oci_rebase_image rule

### PR DESCRIPTION
**Pull request by abayer**
This rule will allow for taking an existing OCI image and replacing the given base image with a new base image, without needing to rebuild all of the other layers.

### CODE REVIEW

File: tests/test_images.bzl [hunk 1]
```diff
@@ -8,3 +8,11 @@ def pull_test_images():
         # Latest at "focal" tag
         digest = "sha256:9d6a8699fb5c9c39cf08a0871bd6219f0400981c570894cd8cbea30d3424a31f",
     )
+
+    oci_pull(
+        name = "ubuntu_jammy",
+        registry = "docker.io",
```
Comment (abayer) [inline tests/test_images.bzl:14]: I need to change this to some ghcr.io image, but I don't know how exactly we pushed `ubuntu_focal` in the first place. =)

Comment (abayer) [reply tests/test_images.bzl]: Trying public.ecr.aws/lts/ubuntu:jammy for the moment...

Comment (abayer) [reply tests/test_images.bzl]: Yeah, that didn't work. =)

File: go/pkg/layer/rebase.go [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+	rebasedManifest, rebasedConfig, err := AppendLayers(ctx, store, newBaseImageDesc, layersToAppend, nil, originalImageConfig.Config.Labels, createdTimestamp, originalImageConfig.Config.Entrypoint)
+	if err != nil {
+		return ocispec.Descriptor{}, ocispec.Descriptor{}, fmt.Errorf("appending original image layers onto new base image: %w", err)
+	}
+
+	if rebasedManifest.Annotations == nil {
+		rebasedManifest.Annotations = make(map[string]string)
+	}
+
+	// Add the new base image's name and
```
Comment (avidal) [inline go/pkg/layer/rebase.go:54]: and....what? don't leave us hanging!

**Review by jprobinson — APPROVED** [review]
⚾ 🌮

**Comment by abayer** [comment]
@jprobinson - ok, this is ready for merge whenever you want to do so. =) I am working on a follow-up that'll change things up so that you can either explicitly specify `--old-base` or, if you don't specify that, it'll inspect the `--original` image looking for the `org.opencontainers.image.base.name` and `org.opencontainers.image.base.digest` annotations on the image. If it finds those annotations, it'll pull down that ref/digest and use that as the old base image. But getting that all wired up correctly to make sure we're generating the proper output is taking a bit of work, so it'll take a little time. =)

**Pull request merged**